pkg/install.bzl

```diff
@@ -0,0 +1,167 @@
+        ctx.label.package,
+        # The name of the binary is the name of this target, minus
+        # "_install_script".
+        ctx.label.name[:-len("_install_script")],
+    )
+
+    ctx.actions.expand_template(
+        template = ctx.file._script_template,
+        output = script_file,
+        substitutions = {
+            "##MANIFEST_INCLUSION##": manifest_file.short_path,
+            # This is used to extend the manifest paths when the script is run
+            # inside a build.
+            "##WORKSPACE_NAME##": ctx.workspace_name,
+            "##TARGET_LABEL##": str(Label(binary_label_str)),
+        },
+        is_executable = True,
+    )
+
+    my_runfiles = ctx.runfiles(
+        files = [manifest_file],
+        transitive_files = depset(transitive = files_to_run),
+    )
+
+    return [
+        DefaultInfo(
+            files = depset([script_file]),
+            runfiles = my_runfiles,
+            executable = script_file,
+        ),
+    ]
+
+_pkg_install_script = rule(
+    doc = """Create an executable package installation script.
+
+    The outputs of this rule are a single python script intended to be used as
+    an input to a `py_binary` target.  All files necesary to run the script are
+    included as runfiles.
+    """,
+    implementation = _pkg_install_script_impl,
+    attrs = {
+        "srcs": attr.label_list(
+            mandatory = True,
+            providers = [
+                [PackageFilegroupInfo],
+                [PackageFilesInfo],
+                [PackageDirsInfo],
+                [PackageSymlinkInfo],
+            ],
+            doc = "Source mapping/grouping targets",
+        ),
+        # This is private for now -- one could perhaps imagine making this
+        # public, but that would require more documentation of the underlying
+        # scripts and expected interfaces.
+        "_script_template": attr.label(
+            allow_single_file = True,
+            default = "//private:install.py.tpl",
+        ),
+    },
+    executable = True,
+)
+
+def pkg_install(name, srcs, **kwargs):
+    """Create an installer script from pkg_filegroups and friends.
+
+    This macro allows users to create `bazel run`nable installation scripts
+    using the pkg_filegroup framework.
+
+    For example:
+
+    ```python
+    pkg_install(
+        name = "install",
+        srcs = [
+            # mapping/grouping targets here
+        ],
+    )
+    ```
+
+    Installation can be done by invoking:
+
+    ```
+    bazel run -- //path/to:install
+    ```
+
+    Additional features can be accessed by invoking the script with the --help
+    option:
+
+    ```
+    bazel run -- //path/to:install --help
+    ```
+
+    Args:
+      name: rule name
+      srcs: pkg_filegroup framework mapping or grouping targets
+      **kwargs: common rule attributes
+    """
+
+    _pkg_install_script(
+        name = name + "_install_script",
+        srcs = srcs,
+        **kwargs
```

Comment:
Can you run this through buildifier. I think that will add trailing , to this line and below at 166.

Reply:
I did, and nothing changed.

Empirically, adding arguments after `**kwargs` or similar produces syntax errors.  I would theorize that not putting a comma after **kwargs makes that somewhat clearer to users.